### DISCUSSION TITLE
Support for Django 1.11 on Py2

### DISCUSSION THREAD

**Discussion by djk2** [boxine/django-huey-monitor#50] [Q&A] [answered]
Hi everyone. 
I have question. 
Now I'm working on old project where on production I have Py2 and Django 1.11 + Huey + Redis.
I'm struggling with performance issues on queue.
I try figure out how I can launch some kind of monitoring for huey (monitoring / metrics). 
I would rather not modify system to much (is really huge - dozens tasks).
Upgrade of environment at now is impossible. 
Are you think that `django-huey-monitor` can be launch with Django 1.11 and Py2 or this is completely uncompatible?
I will be grateful for response.

**Comment by phihag** [comment, marked as answer]
Hi @djk2 . I'm sorry, but at the moment we're not planning to support either Python 2 or Django 1.11.

As Django 1.11 security support ended April 1st 2020, this would be an insecure configuration anyways.
We also use Python 3 features extensively.

**Reply by djk2** [reply]
Hi @phihag  thx for quick response. 
I understand that you are not planning exend code for back-support :) 
I would rather  asked about your opinion about is sense traying launch `django-huey-monitor` on Dj 1.11 and Py2, there is sense or not because for example you at now know that wont be work.

**Reply by phihag** [reply]
Looking through the code, there's nothing fundamental that would prevent running on Python 2.x.
It won't work unmodified, since `functools.lru_cache` would be missing. But that seems to be it.

You could monkeypatch, even with a simple `import functools; functools.lru_cache = lambda func: func`.

In a similar vein, Django 1.11 may require some minor import adjustments, but nothing fundamental.

But as I wrote above, I cannot in good conscience recommend such a setup. I would strongly recommend addressing technical debt first by upgrading to Python 3 and Django 3.

**Reply by djk2** [reply]
@phihag Thx for help. 
At now I will try use `Huey Prometheus Exporter`. 
Maybe I will back to `django-huey-monitor` if I have more time. 
Now we strongly working on upgrading app to Py3 but still django is on 1.11.
